### ISSUE TITLE
problem loading rrelaxiv.so following install of Mac binary via drat

### ISSUE THREAD

**Issue by benthestatistician** [josherrickson/rrelaxiv#3]
After `drat`-assisted install on my Intel Mac Mini,  I'm seeing

```
library(rrelaxiv)
Error: package or namespace load failed for ‘rrelaxiv’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/bbh/Library/R/x86_64/4.2/library/rrelaxiv/libs/rrelaxiv.so':
  dlopen(/Users/bbh/Library/R/x86_64/4.2/library/rrelaxiv/libs/rrelaxiv.so, 0x0006): Symbol not found: __gfortran_os_error_at
  Referenced from: <97A6A120-3DD9-3C70-8FAD-A89905599DCC> /Users/bbh/Library/R/x86_64/4.2/library/rrelaxiv/libs/rrelaxiv.so
  Expected in:     <5B0F7278-9D16-3B60-A413-59EEF538E568> /usr/local/gfortran/lib/libgfortran.5.dylib
```

Session details, from `sessioninfo::session_info()`:
```
─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.3 (2023-03-15)
 os       macOS Big Sur ... 10.16
 system   x86_64, darwin17.0
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/Detroit
 date     2023-08-08
 pandoc   2.19.2 @ /opt/local/bin/pandoc

─ Packages ───────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 cli           3.6.0   2023-01-09 [1] CRAN (R 4.2.0)
 sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.2.0)

 [1] /Users/bbh/Library/R/x86_64/4.2/library
 [2] /Library/Frameworks/R.framework/Versions/4.2/Resources/library
 [3] /Library/Frameworks/R.framework/Versions/4.2/Resources/miworkspace-library

──────────────────────────────────────────────────────────────────────────────
```

I am able to install from github (v0.2.1) using devtools and the instructions in the README.

**Comment by josherrickson** [comment]
I can try rebuilding the binary for 4.2 on intel mac. I probably should also add a 4.3 intel mac, which I notice we don't have.

Looking online, I'm seeing some discussion that this error could be related to multiple install of gfortran. Is it possible you have more than one installed?

**Comment by benthestatistician** [comment]
It's quite possible that I had multiple gfortran's.  You're saying that could be an issue even for drat-assisted installs? 

Let's see if we can figure this out before you start rebuilding binaries.  If the real issue is multiple gfortran's, then our time would be better spent adding something about that to the README for this repo.

**Comment by benthestatistician** [comment]
I did have an out-of-date gfortran installed when I attempted this. Unfortunately, removing it didn't seem to fix the problem. I still get

```
> library("rrelaxiv")
Error: package or namespace load failed for ‘rrelaxiv’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/bbh/Library/R/x86_64/4.2/library/rrelaxiv/libs/rrelaxiv.so':
  dlopen(/Users/bbh/Library/R/x86_64/4.2/library/rrelaxiv/libs/rrelaxiv.so, 0x0006): Symbol not found: __gfortran_os_error_at
  Referenced from: <97A6A120-3DD9-3C70-8FAD-A89905599DCC> /Users/bbh/Library/R/x86_64/4.2/library/rrelaxiv/libs/rrelaxiv.so
  Expected in:     <5B0F7278-9D16-3B60-A413-59EEF538E568> /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libgfortran.5.dylib
```

**Comment by josherrickson** [comment]
Can you make sure `which gfortran` is pointing to a valid version? What does `gfortran --version` return? What about `gcc --version`?

**Comment by josherrickson** [comment]
Also, does `/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libgfortran.5.dylib` exist for you? If not, can you see if the file exists at a slightly different location?

**Comment by benthestatistician** [comment]
1. I should have made clear that I was attempting to install the prefab binary, in an R that I got as binary from the CRAN installer (rather than an R I had built from source). 
2. When I ran these experiments, `gfortran` wasn't pointing to anything (`which gfortran` gave nothing). Since then I've set up gcc12.2.0_2 via Macports, and it points to that.
3. I do (and did) have `/Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libgfortran.5.dylib`
4. On another computer I installed it from source, after also installing gcc12.2 and R 4.3 from source. I'll soon do the same on this computer, committing to keeping that gcc around, unless you suggest I do otherwise in order to troubleshoot this further. (12.2 is the right gcc for R version 4.3 but not for R 4.2.)

**Comment by josherrickson** [comment]
You shouldn't need to install R from source, but you should need gcc installed. If you've installed that, try re-installing from drat.

**Comment by benthestatistician** [comment]
I did have gcc at the time that I tried the install.  However, it probably wasn't same gcc version the R 4.2.3 distro would have been compiled with, and it may also have been out of step with my version of the Mac OS. 

Unfortunately I don't see any good ways to get that version of gcc (8.2) running on Ventura: I can't find a suitable standalone installer; macports has a gcc8, but it's 8.5 not 8.2.  (Plus, when I try to install it I get "gcc8 is known to fail. Try to install anyway?" for it and the various packages it depends on.)   So I'm afraid I'll have to leave the experiments with further gcc's at that.  

OTOH, I now have gcc working and aligned with my R 4.3 -- I could try troublshooting an R 4.3 binary, if that's helpful.
